clippy_lints/src/casts/cast_lossless.rs

```diff
@@ -25,7 +25,14 @@ pub(super) fn check(
     // The suggestion is to use a function call, so if the original expression
     // has parens on the outside, they are no longer needed.
     let mut applicability = Applicability::MachineApplicable;
-    let opt = snippet_opt(cx, cast_op.span);
+    let opt = snippet_opt(
+        cx,
+        if cast_op.span.from_expansion() {
+            cast_op.span.source_callsite()
+        } else {
+            cast_op.span
+        },
+    );
```

Comment:
```suggestion
    let opt = snippet_opt(cx, cast_op.span.source_callsite());
```
Small nit, but, I don't think the from_expansion check is needed. `source_callsite()` already does the check and returns the same span if it's not from an expansion. Or is there some other reason it's done like that?